plan/logical_plan_builder.go

```diff
@@ -647,7 +655,7 @@ func (b *planBuilder) buildProjection4Union(u *LogicalUnionAll) {
 			dstType := unionSchema.Columns[i].RetType
 			srcType := srcCol.RetType
 			if !srcType.Equal(dstType) {
-				exprs[i] = expression.BuildCastFunction(b.ctx, srcCol, dstType)
+				exprs[i] = expression.BuildCastFunction4Insert(b.ctx, srcCol, dstType)
```

Comment:
Seems like `BuildCastFunction4Insert` is only used for `Union`, maybe name the function to `BuildCastFunction4Union` is better?

Reply:
Emma, but maybe it will use for insert, insert cast logic is the same as this one, but now insert values is datum based cast, or we change to insert in future?

Reply:
We can change it in the future. I believe it will take us a long time to reach that future 😂